From 2bc81607a7710d4c07f41eb4e9fe1d3309afe19e Mon Sep 17 00:00:00 2001
From: Seth Forshee <seth.forshee@canonical.com>
Date: Mon, 15 Jun 2020 15:16:11 -0500
Subject: [PATCH 24/35] UBUNTU: SAUCE: shiftfs -- Fix build errors from missing
 fiemap definitions

shiftfs FTBFS with 5.8-rc1:

 /tmp/kernel-sforshee-6727637082e4-45IQ/build/fs/shiftfs.c: In function 'shiftfs_fiemap':
 /tmp/kernel-sforshee-6727637082e4-45IQ/build/fs/shiftfs.c:731:13: error: dereferencing pointer to incomplete type 'struct fiemap_extent_info'
 /tmp/kernel-sforshee-6727637082e4-45IQ/build/fs/shiftfs.c:731:26: error: 'FIEMAP_FLAG_SYNC' undeclared (first use in this function); did you mean 'FS_XFLAG_SYNC'?

It seems that shiftfs was getting linux/fiemap.h included
indirectly before. Include it directly.

Signed-off-by: Seth Forshee <seth.forshee@canonical.com>
---
 fs/shiftfs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/shiftfs.c b/fs/shiftfs.c
index 5d88193b41db..f9a5c94a9793 100644
--- a/fs/shiftfs.c
+++ b/fs/shiftfs.c
@@ -20,6 +20,7 @@
 #include <linux/posix_acl.h>
 #include <linux/posix_acl_xattr.h>
 #include <linux/uio.h>
+#include <linux/fiemap.h>
 
 struct shiftfs_super_info {
 	struct vfsmount *mnt;
-- 
2.30.2

